tcpmaxconn/tcpmaxconn.go

```diff
@@ -27,11 +26,11 @@ func connection_handler(id int, host string, port int, wg *sync.WaitGroup) {
 			fmt.Println(str)
 		}
 		if err!= nil {
-			break
+			wg.Done()
+			return err
 		}
 	}
-	fmt.Println("\t runner "+strconv.Itoa(id)+" got its connection closed")
-	wg.Done()
+	// fmt.Println("\t runner "+strconv.Itoa(id)+" got its connection closed")
```

Comment:
I'd move this line to the "if err" block rather than commenting it

Reply:
👍